src/main/java/com/minecolonies/coremod/commands/colonycommands/DisableBarbarianSpawnsCommand.java

```diff
@@ -43,35 +53,63 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     }
 
     @Override
-    public void execute(
-                         @NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final List<ActionArgument> actionArgumentList,
+            @NotNull final Map<String, Object> argumentValueByActionArgumentNameMap) throws CommandException
     {
+        Colony colony = null;
+        final Object colonyObject = argumentValueByActionArgumentNameMap.get("colony");
+        if (null != colonyObject)
+        {
+            colony = (Colony) colonyObject;
+        }
 
-        if (sender instanceof EntityPlayer && !isPlayerOpped(sender))
+        boolean canHaveBarbEvents = false;
+        final Object disableSpawnsObject = argumentValueByActionArgumentNameMap.get("disableSpawns");
+        if (null != disableSpawnsObject)
         {
-            sender.sendMessage(new TextComponentString("Must be OP to use this command"));
+            final Boolean disableSpawnsBoolean = (Boolean) disableSpawnsObject;
+            canHaveBarbEvents = !disableSpawnsBoolean.booleanValue();
         }
 
-        final int colonyId;
+        executeShared(server, sender, colony, canHaveBarbEvents);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "executeShared()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)